02-oop-inheritance/javaScript/refactored_JS/downloableMovie v0.1.js

```diff
@@ -0,0 +1,94 @@
+var MovieObserver = function ( ) {};
+MovieObserver.prototype.playing = function ( title ) {
+    console.log('A movie with title ' + title + ' has been played');
+};
+MovieObserver.prototype.stopped = function( title ) {
+    console.log('A movie with title ' + title + ' stopped');
+};
+
+var Movie = function ( paramName, paramValue ) {
+  //Variables
+  this.attributes = [];
+  this.attributes[paramName] = paramValue; 
+  this.observer = new MovieObserver();
+};//end movie constructor
+
+Movie.prototype = {
+  
+  get: function (atribute) {
+     return this.attributes[atribute];
+  }, 
+
+  set:function (atribute, value) {
+    this.attributes[atribute] = value;
+      
+  },
+
+  play:function () {
+    console.log('This is Movie.play method: I am a movie. This is my title: ' + this.attributes['title']);
+    this.observer.playing(this.attributes['title']);
+  },
+
+  stop:function () {
+    this.observer.stopped(this.attributes['title']);
+  }
+
+};
+
+var DownloableMovie  = function ( paramName, paramValue ) {
+  Movie.call(this, paramName,paramValue);
+};
+
+var inheritPrototype = function (paramParentObject, paramChildObject ) {
+  var copyOfParent =Object.create(paramParentObject.prototype);
+  copyOfParent.constructor = paramChildObject;
+  paramChildObject.prototype = copyOfParent;
+};
+
+
+
+inheritPrototype(Movie,DownloableMovie);
+
+/**
+*Augment existing 'class' with a method from another.
+*This function is courtesy of Addy Osmani
+* From the book JavaScript Design Patterns
+*/
+var augment= function ( receivingClass, givingClass ) {
+  // only provide certain methods
+  if ( arguments[2] ) {
+    for (var i=2, len=arguments.length; i<len; i++) {
```

Comment:
Take care of hoisting here, also try to be more specific with the variables name in the future.